### ISSUE TITLE
ocaml 5.1.0 changes test results

### ISSUE THREAD

**Issue by avsm** [ocaml-multicore/ocaml-uring#94]
OCaml 5.1.0 changes the rendering of weak row variables (https://github.com/ocaml/ocaml/pull/12107) to make the type variable more explicit. It means the uring tests don't work on 5.1.0 right now, so we need to constrain them away on older releases and probably make the tests 5.1.0+ 

```
 # let a, b = Unix.(socketpair PF_UNIX SOCK_STREAM 0);;
 val a : Unix.file_descr = <abstr>
 val b : Unix.file_descr = <abstr>
 # let bufs = [Cstruct.of_string "hi"];;
 val bufs : Cstruct.t list = [{Cstruct.buffer = <abstr>; off = 0; len = 2}]
 # Uring.send_msg t a ~fds:[r; w] bufs `Send;;
-- : _[> `Send ] Uring.job option = Some <abstr>
+- : ([> `Send ] as '_weak24) Uring.job option = Some <abstr>
 # Uring.submit t;;
 - : int = 1
 # consume t;;
-- : _[> `Send ] * int = (`Send, 2)
+- : ([> `Send ] as '_weak24) * int = (`Send, 2)
 # let recv_buf = Cstruct.of_string "XX";;
 val recv_buf : Cstruct.t = {Cstruct.buffer = <abstr>; off = 0; len = 2}
 # let recv = Uring.Msghdr.create ~n_fds:2 [recv_buf];;
@@ -662,11 +662,11 @@ val recv : Uring.Msghdr.t = <abstr>
 # List.length (Uring.Msghdr.get_fds recv);;
 - : int = 0
 # Uring.recv_msg t b recv `Recv;;
```

**Comment by patricoferris** [comment]
Fixed in #96

**Comment by avsm** [comment]
Thanks!
